partition/roles/mgmt-firewall/README.md

```diff
@@ -0,0 +1,120 @@
+# Automated Firewall Setup with Ansible
+
+This role automates the configuration of management firewalls using Ansible. It is designed to streamline the process of setting up firewalls for consistent deployment before mounting devices in the data center. By utilizing default configurations and flexible variables, this role simplifies the setup across multiple devices.
+
+**Note**: This role is intended to be run on devices reset to factory defaults.
+
+## Supported Devices
+
+| Manufacturer | Model  |
+| ------------ | ------ |
+| Teltonika    | RUTXR1 |
+
+## Key Features
+
+- **Automated firewall setup** using default configurations
+- **VLAN and BGP** configuration support
+- **Dynamic port forwarding** setup
+- **Pre-configured firewall rules** for LAN, WAN, and global settings
+- **Device-specific customization** via `routers.yaml`
+
+## Prerequisites
+
+- The device must be **reset to factory defaults** before running this role.
+- An initial login is required to change the root password using credentials defined in the `routers.yaml` file.
+
+## Configuration Details
+
+### Firewall Rules
+
+The firewall is configured with the following settings by default:
+
+1. **Global Settings:**
+
+   - Drop invalid packets: **Enabled**
+   - Input: **Drop**
+   - Output: **Accept**
+   - Forward: **Drop**
+   - Offloading: **On**
+
+2. **LAN Configuration:**
+
+   - Input, Output, Forward: **Accept**
+   - Masquerading: **On**
+   - MSS Clamping: **On**
+
+3. **WAN Configuration:**
+   - Input: **Drop**
+   - Output: **Accept**
+   - Forward: **Drop**
+   - Masquerading: **On**
+   - MSS Clamping: **On**
+
+### VLAN Configuration
+
+- **VLAN 1:** Tagged to port 4
+- **VLAN 2:** Tagged to port 5 (WAN)
+- Other VLANs can be configured dynamically.
+
+### BGP Configuration
+
+- The BGP peer is **hardcoded** as `mgmtsrv`.
+- The IP address and AS number can be configured dynamically.
+
+## Interfaces
+
+Both LAN and WAN interfaces share the following mandatory fields:
+
+| Field          | Description    |
+| -------------- | -------------- |
+| `name`         | Interface name |
+| `ipaddr`       | IP address     |
+| `netmask`      | Subnet mask    |
+| `device`       | Router port    |
+| `dhcp_options` | (LAN Only)     |
+| `metric`       | (WAN Only)     |
+| `gateway`      | (WAN Only)     |
+| `dns` (List)   | (WAN Only)     |
```

Comment:
We have a dotted-syntax for dictionary fields. Maybe you can take a look somwhere here?

https://github.com/metal-stack/metal-roles/blob/master/partition/roles/dhcp/README.md?plain=1#L12-L16

Reply:
These are dic fields under mgmt_firewall_lan.
What exactly do you mean?

```yaml
mgmt_firewall_lan:
    - name: ''
      ipaddr: ''
      netmask: ''
      device: ''
      dhcp_options:
        - { option: '3', value: '' }
        - { option: '6', value: '' }
        - { option: '12', value: '' }
 ```

Reply:
Yes, can you put this into the variables table with `mgmt_firewall_lan.name`, `mgmt_firewall_lan.ipaddr`? Then we have it similar across our roles.